.github/workflows/next-lint-check.yml

```
@@ -0,0 +1,31 @@
+name: Next.js Linting Check
+
+on:
+  pull_request:
+    branches:
+      - main
+      - master
+  push:
+    branches:
+      - main
+      - master
+
+jobs:
+  lint:
+    name: Linting with Next.js
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v2
+
+      - name: Use Node.js
+        uses: actions/setup-node@v3
```

```suggestion
        uses: actions/setup-node@v4
```
Latest is [v4](https://github.com/actions/setup-node/releases/tag/v4.0.0), it should not break anything as far as I can see...

---

I believe here as well lets go with the `v3` to prevent new bugs from `v4`?